### PR TITLE
token tests p1

### PR DESCRIPTION
**Pull request by eventhorizonpl**
Hi,

This PR adds some tests for token program.

Best regards,
Michal

### CODE REVIEW

File: token/program/src/state.rs [hunk 1]
```diff
@@ -286,3 +286,93 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
...
+        assert_eq!(
+            mint,
+            solana_program::program_error::ProgramError::InvalidAccountData
+        );
+    }
+
+    #[test]
+    fn test_account_state() {
+        let account_state = AccountState::Initialized;
+        assert_eq!(account_state, AccountState::Initialized);
```
Comment (joncinque) [inline token/program/src/state.rs:312]: Was this supposed to test something else?

File: token/program/src/state.rs [hunk 2]
```diff
@@ -286,3 +286,93 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
...
+        let src: [u8; 82] = [0; 82];
+        let mint = Mint::unpack_from_slice(&src).unwrap();
+        assert!(!mint.is_initialized);
+
+        let mut src: [u8; 82] = [0; 82];
+        src[45] = 2;
+        let mint = Mint::unpack_from_slice(&src).unwrap_err();
+        assert_eq!(
+            mint,
+            solana_program::program_error::ProgramError::InvalidAccountData
```
Comment (joncinque) [inline token/program/src/state.rs:305]: nit: program error is already imported earlier, so these can be
```suggestion
           ProgramError::InvalidAccountData
```

File: token/program/src/state.rs [hunk 3]
```diff
@@ -286,3 +286,93 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
...
+            multisig,
+            solana_program::program_error::ProgramError::InvalidAccountData
+        );
+    }
+
+    #[test]
+    fn test_unpack_coption_key() {
+        let src: [u8; 36] = [0; 36];
+        let result = unpack_coption_key(&src).unwrap();
+        assert_eq!(result, solana_program::program_option::COption::None);
```
Comment (joncinque) [inline token/program/src/state.rs:348]: nit:
```suggestion
        assert_eq!(result, COption::None);
```

**Review by joncinque — COMMENTED** [review]
These are really great! Just a couple of tiny nits, then we can integrate this

**Comment by eventhorizonpl** [comment]
@joncinque Fixed. Thank you for the review.

**Comment by joncinque** [comment]
Thanks for the contribution, looks good!

**Pull request merged**